website/docs/reference/resource-properties/constraints.md

```diff
@@ -518,3 +518,73 @@ alter table schema_name.my_model add constraint 472394792387497234 check (id > 0
 </div>
 
 </WHCode>
+
+### Custom constraints on models for advanced configuration of tables
+
+In dbt Cloud and dbt Core, you can use custom constraints on models for the advanced configuration of tables. Different data warehouses support different syntax and capabilities. However, custom constraints allow you to add configuration to specific columns.
+
+  - For example, custom constraints allow you to set [masking policies](https://docs.snowflake.com/en/user-guide/security-column-intro#what-are-masking-policies) in Snowflake when using a Create Table As Select (CTAS).
+  
+  - [Databricks](https://docs.databricks.com/en/sql/language-manual/sql-ref-syntax-ddl-create-table-using.html) and [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/data-definition-language#column_name_and_column_schema) are other examples of data warehouses having their own set of parameters that can be set for columns in their CTAS statements.
+
+
+You can implement constraints in a couple of different ways:
+
+- [Custom constraints with tags](#custom-constraints-with-tags)
+- [Custom constraints without tags](#custom-constraints-without-tags)
+### Custom constraints with tags
+
+Here's an example of how to implement tag-based masking policies with contracts and constraints using the following syntax:
+
+<File name='models/constraints_example.yml'>
+
+```yaml
+
+models:
+  - name: my_model
+    config: null
+      contract:
+        enforced: true
+      materialized: table
```

Comment:
materialized should be nested under contract
```suggestion
        materialized: table
```

Reply:
@b-per 's example here has the right yaml format: https://github.com/dbt-labs/dbt-adapters/issues/85#issuecomment-2314547162